packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,708 @@
+# React Grid - Custom Plugin Development
+
+
+## Table of Contents
+1. [Introduction](#introduction)
+2. [Render Custom Markup](#render-custom-markup)
+3. [Share Values Between Plugins](#share-values-between-plugins)
+4. [Use the Functionality of Built-In Plugins](#use-the-functionality-of-built-in-plugins)
+5. [Final Steps](#final-steps)
+
+
+## Introduction
+
+This tutorial is aimed at advanced users who are familiar with [basic Grid concepts](./getting-started.md) and who want to add new functionality to the Grid by using the [React Core](../../../core/docs/guides/fundamentals.md) components.
+
+In our particular case, we will extend the Grid's toolbar with a select box that will allow users to filter the grid by a required column. The purpose of the tutorial is not to implement a very useful feature. The main idea is to show how to add to the Grid component functionality that is not provided out of the box.
+
+The initial code is shown below. As you can see, we will use the Grid in the `Material-UI` theme.
+
+*index.js:*
+```jsx
+import {
+  Grid,
+  Table,
+  TableHeaderRow,
+  Toolbar
+} from '@devexpress/dx-react-grid-material-ui';
+import service from './data.js';
```

Comment:
The `rows: service.getCustomers()` code below looks a bit strange to me since I expect an async call here. So I would replace this with a simpler code:

```js
import customers from './data'
...
{
  ...
   rows: customers
}
```

Reply:
Makes sense. I'll fix it.